### PR TITLE
Update buck (merge facebook/buck master)

### PR DESCRIPTION
**Pull request by qyang-nj**
* Merge `facebook/buck` master
* update Skylark syntax to stop buck from yelling

#76 should be fixed by the latest buck

### CODE REVIEW

File: Config/configs.bzl [hunk 1]
```diff
@@ -1,5 +1,11 @@
 load("//Config:utils.bzl", "config_with_updated_linker_flags", "configs_with_config")
 
+def merge_dict(a, b):
```
Comment (dfed) [inline Config/configs.bzl]: 🚀

**Comment by qyang-nj** [comment]
> [2019-04-11 00:06:50.641][error][command:1f77b179-29b2-4923-a08a-79f18fe9e584][tid:27][com.facebook.buck.apple.SwiftStdlibStep] Error running [/Applications/Xcode-9.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo, -archs, buck-out/gen/App/ExampleApp#dwarf-and-dsym,no-include-frameworks/ExampleApp.app/ExampleApp]: Optional[fatal error: /Applications/Xcode-9.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: unknown flag: -archs

It seems [this commit](https://github.com/facebook/buck/commit/a2d02890d5ee68a5aadee9fcdaecdc37e97876f5) breaks Xcode 9.3 (and older) toolchain. We should update to Xcode 10 on CI.

**Comment by dfed** [comment]
Absolutely agree. We’ll need to update our `.travis.yml` file

**Comment by qyang-nj** [comment]
Getting error:
> You're using Java 12, but Buck requires Java 8.

We ran into a dilemma with Xcode 10 and Buck on Travis.
> While Mac jobs can test against multiple JDK versions using the jdk key, macOS images up to xcode9.3 can only switch up to Java 8, and images xcode9.4 and later can switch to Java 10 (if pre-installed) and later. In practical terms, if your Mac build requires Java 8 and below, use xcode9.3 (or below); if your build requires Java 10 and later, use xcode9.4 (or later).  

https://docs.travis-ci.com/user/reference/osx/#jdk-and-macos

**Comment by dfed** [comment]
Unknown

We may be diving down quite the rabbit hole here, but: https://github.com/facebook/buck/issues/1548 and https://github.com/facebook/buck/issues/2037 seem somewhat promising?

**Comment by qyang-nj** [comment]
Close this one in favor of #108

**Pull request closed without merging**